components/scripts/gradle/05-validate-remote-build-caching-ci-local.sh

```diff
@@ -369,11 +384,30 @@ EOF
   print_interactive_text "${text}"
 }
 
+collect_remote_build_cache() {
+  collect_remote_build_cache_type
+  collect_remote_build_cache_url
+}
+
+collect_remote_build_cache_type() {
+  local default_remote_cache_type="<project default>"
+  prompt_for_setting "What is the remote build cache type to use? [develocity, gradle-enterprise, http, or <BLANK>]" "${remote_build_cache_type}" "${default_remote_cache_type}" remote_build_cache_type
+
+  if [[ -n "${remote_build_cache_type}" && "${remote_build_cache_type}" != 'http' && "${remote_build_cache_type}" != 'gradle-enterprise' && "${remote_build_cache_type}" != 'develocity' ]]; then
+    print_bl
+    die "ERROR: Invalid value for remote build cache type. Values are 'develocity', 'gradle-enterprise', 'http', or <BLANK> for project default." "${INVALID_INPUT}"
```

Comment:
It is not very user-friendly to completely exit the script in interactive mode if the user enters an invalid value. It would be better to print the error message, and call `collect_remote_build_cache_type()` again (or do the validation   using a do...while loop) until the user does provide a valid value.

Reply:
I also wonder if we should include a link to some documentation in the error message so that users who don't know what this is can learn more about it.

Reply:
> do the validation using a do...while loop

@etiennestuder and I discussed exactly this and decided to move forward with just exiting for now. It comes at the cost of user-friendliness, but it's the most straight forward approach. If it turns out to be problematic for users we can certainly fix it.

> I also wonder if we should include a link to some documentation in the error message so that users who don't know what this is can learn more about it.

That's not a bad idea, but would require documentation to link to. 😅 

I've created issue https://github.com/gradle/develocity-build-validation-scripts/issues/809 for this so that it can be addressed in a subsequent PR.